src/context/cdhashmap.h

```diff
@@ -180,7 +180,7 @@ class CDOhash_map : public ContextObj {
          bool allocatedInCMM = false) :
     ContextObj(allocatedInCMM, context),
     d_key(key),
-    d_data(),
+    d_data(data),
```

Comment:
I would not include this change without *significant* testing (a significant fraction of SMT-LIB in multiple theories) or someone really thinking this through. It introduces a very subtle change in the object that is probably ok, but it is hard to be sure. The issue would be the final destructor call when !atLevelZero would previously have been backtracked to Data() whereas now d_data can linger until the destructor for the CDOhash_map happens. It is a reasonable change, but it needs more testing.

Reply:
I'll fire up a StarExec run for that. Would incremental benchmarks be a better choice since they use context dependent data structures more extensively?

Reply:
I would use a mix of incremental and non-incremental with a bias towards a lot of theories and combinations.

Reply:
I am running a debug build of this branch with the competition run script on all incremental benchmarks and 10'000 random benchmarks from SMT-LIB. Timeout is 5min. Will report back here once the runs complete.

Reply:
SG I'll be interested in the results.

Reply:
I finally got around to do the benchmarks (10'000 random SMT-LIB benchmarks and all incremental benchmarks) and look at the results and I could not see any additional failures (production build, assertions enabled).

Reply:
Thanks for checking this!